ts/Core/Series/Series.ts

```diff
@@ -2302,9 +2302,10 @@ class Series {
             // some API data
             point.category = (
                 categories &&
-                typeof (categories as any)[point.x as any] !== 'undefined' ?
-                    (categories as any)[point.x as any] :
-                    point.x
+                point.x !== null &&
```

Comment:
Are there any actual cases where `point.x` is `null`, or does this test come from the typing? I suspect it mey be a relic of the past, and that we instead should do `isNumber(point.x)` to rule out `undefined`.

Reply:
This type is coming from MockPoint: https://github.com/highcharts/highcharts/blob/master/ts/Extensions/Annotations/MockPoint.ts#L494-L508

Reply:
Neither works:
```ts
point.category = categories && pick(categories[point.x], point.x);
point.category = pick(categories && categories[point.x], point.x);
```
> 3 FAILED, 1910 SUCCESS

Reply:
Ok... Then I suspect something must be wrong with the typing. I'll check.

Reply:
Confirmed that the typing of `axis.categories` is wrong. It can be a boolean, so in the failing cases it was `false`. Pushing a commit now.